Source/DafnyCore/Compilers/Dafny/Compiler-dafny.cs

```diff
@@ -1796,7 +1796,11 @@ private class ExprLvalue : ILvalue {
         EmitExpr(expr.Initializer, inLetExprBody, new BuilderSyntaxTree<ExprContainer>(elem), wStmts);
 
         var n = new ExprBuffer(null);
-        EmitExpr(expr.N, inLetExprBody, new BuilderSyntaxTree<ExprContainer>(n), wStmts);
+        var size = expr.N;
+        if (size.Type.AsNewtype is { }) {
```

Comment:
This is untested for now but will be soon as I'm writing the Rust compiler that leverages this experimental compiler.

Reply:
I assume that means the test file doesn't work for Rust for an unrelated reason?

Reply:
Indeed. This part is totally untested for now.